File: framework/java/src/main/java/soda/web/resp/BaseResponse.java
package soda.web.resp;

import com.sun.net.httpserver.Headers;

abstract class BaseResponse implements Response {

    private static final String HEADER_CONTENT_TYPE = "Content-Type";
    
    private final int httpCode;

    private Headers headers = new Headers();

    public BaseResponse(int httpCode, String contentType) {
        this.httpCode = httpCode;
        headers.set(HEADER_CONTENT_TYPE, contentType);
    }

    @Override
    public int getHttpCode() {
        return httpCode;
    }

    @Override
    public Headers getHeaders() {
        return headers;
    }

    @Override
    public String getContentType() {
        return headers.getFirst(HEADER_CONTENT_TYPE);
    }

}

File: framework/scala/src/main/scala/soda/scala/web/resp/BaseResponse.scala
package soda.scala.web.resp

import com.sun.net.httpserver.Headers

abstract class BaseResponse(private val httpCode: Int, contentType: String) extends Response {

  private val headers = new Headers()

  headers.set("Content-Type", contentType)

  override def getHttpCode: Int = httpCode

  override def getHeaders: Headers = headers

  override def getContentType: String = headers.getFirst("Content-Type")

}
